src/widgets/table/table.rs

```diff
@@ -1484,7 +1364,264 @@ mod tests {
                 .rows(vec![])
                 .footer(Row::new(vec!["h", "i"]))
                 .column_spacing(0);
-            assert_eq!(table.get_columns_widths(10, 0), &[(0, 5), (5, 5)])
+            assert_eq!(table.get_columns_widths(10, 0), [(0, 5), (5, 5)])
+        }
+
+        #[test]
+        fn spacers_priority_unconstrained() {
```

Comment:
For a reader looking at this code for the first time, neither priority or unconstrained have any meaning with respect to the wider ratatui context, and so are floating undefined abstract concepts. Choosing a test name that is more grounded, defined, concrete would help make it obvious what this test is testing.

Reply:
Good idea. I've updated the test name now.

Reply:
This still introduces two concepts that aren't actually defined anywhere.
Spacers - is a layout concept, but doesn't appear in the public API of Table at all, so it's not clear how this relates.
Priority - many of the comments introduce or reference priority, but the reference is single sided comparison e.g. "highlight_symbol and spacing are prioritized". Prioritized compared to what?

Reply:
Trying these names:

- excess_area_highlight_symbol_and_column_space_allocation
- insufficient_area_highlight_symbol_and_column_space_allocation
- insufficient_area_highlight_symbol_allocation_with_no_column_spacing

Reply:
what is the verb? (i.e. the behavior that is being tested)

Reply:
We are testing that "highlight_symbol and column spaces are allocated" in three cases

- when there's excess area to do so
- when there's insufficient area to do so

We are also testing that highlight_symbol is allocated when column spacing is zero

- when there's insufficient area to do so

Reply:
- renders_columns_with_column_spacing_and_enough_space
- renders_columns_with_column_spacing_and_insufficient_space
- renders_columns_with_no_column_spacing_and_insufficient_space

Reply:
If you're leaving the multiple tests per function the "how does this render the columns" part becomes the comment as well as the "with these extra relevant constraints that modify the behavior"

Reply:
Specifically we are checking that highlight spacing is working the way we expect it to. Your names don’t have anything about highlight spacing.

Reply:
- renders_highlight_spacing_when...